dailymed-ui/src/components/RxTermsSearch/index.jsx

```diff
@@ -97,46 +63,87 @@ export default class RxTermsSearch extends React.Component {
+                //Text shown in the dropdown box when a search is being fetched
+                searchText="Searching..."
+                placeholder="Enter a drug name..."
+                //Whether the dropdown should show the loading icon
+                isLoading={this.state.isLoading}
+                //Runs when any text is being typed into the typeahead
+                onSearch={(text) => {
+                  this.setState({ isLoading: true });
+                  this.searchForTerms(text, settings, options);
+                }}
+              />
+              <Form.Text>
+                Minimum {settings.drugNameMinLength} characters.
+              </Form.Text>
+            </Form.Group>
+
+            <Form.Group>
+              <Form.Label>Drug strength and form</Form.Label>
+              <Typeahead
+                id="drug-strength-typeahead"
+                minLength={0}
+                highlightOnlyResult={settings.highlightOnlyResult}
+                paginate={settings.paginate}
+                maxResults={settings.maxResults}
+                labelKey={"strengthAndForm"}
+                onInputChange={(text) =>
+                  this.handleDrugStrengthAndFormChange(text)
+                }
+                onChange={(selection) => this.setDrugStrengthAndForm(selection)}
+                options={options.drugStrengthAndForm}
+                placeholder="Choose a strength and form"
+              />
+            </Form.Group>
+
+            <div className="float-right">
+              <Button variant="light">Reset</Button>{" "}
+              <Button variant="primary" disabled={!selected.rxcui}>
+                Next
+              </Button>{" "}
+            </div>
+          </div>
         );
     }
+    /** Adds queried drug names and strengths to the typeahead options list and empties the selected array.
+     * @param {string} text - The current text entered into the input
+     * @param {object} settings - The settings for the typeahead
+     * @param {object} options - The options for the typeahead dropdown
+     * 
+    */
+    searchForTerms(text, settings, options) {
+        if (text.length >= settings.drugNameMinLength)
+            fetch(
+                `https://clinicaltables.nlm.nih.gov/api/rxterms/v3/search?ef=DISPLAY_NAME,STRENGTHS_AND_FORMS,RXCUIS&maxList=&terms=${text}`
+            )
+                .then((resp) => resp.json())
+                .then((json) => {
+                    //** Deconstruct the fetch result */
+                    const { DISPLAY_NAME: names, STRENGTHS_AND_FORMS: forms, RXCUIS: cuis } = json[2];
+                    //**If names are returned, set the dropdown options to the returned names */
+                    if (names && forms)
+                        options.drugName = names.map((name, index) => {
+                            return {
+                                displayName: name,
+                                strengthsAndForms: forms[index]
```

Comment:
@davidhicks980 - you're missing a key piece of this function - we need to also add RXCUIs to this object so we can use them later once both drug name and drug strength/form are selected.

Reply:
It has to be nested like this:

displayName
strengthsAndForms
    strengthAndForm
    rxcui
    ...
    strengthAndForm
    rxcui

Reply:
I don't know WHY this works with the weird `[index][index]` but it seems to work...

```
                .then((json) => {
                    //** Deconstruct the fetch result */
                    const { DISPLAY_NAME: names, STRENGTHS_AND_FORMS: forms, RXCUIS: cuis } = json[2];
                    //**If names are returned, set the dropdown options to the returned names */
                    if (names && forms)
                        options.drugName = names.map((name, index) => {
                            return {
                                displayName: name,
                                strengthsAndForms: forms[index].map((form, index) => {
                                  return {
                                    strengthAndForm: form,
                                    rxcui: cuis[index][index]
                                  };
                                })
                            };
                        });

                    this.setState({ isLoading: false, options, selected: [] });
                });
```

Reply:
This is kind of ugly but seems to work... thoughts?

```
                .then((json) => {
                    //** Deconstruct the fetch result */
                    const { DISPLAY_NAME: names, STRENGTHS_AND_FORMS: forms, RXCUIS: cuis } = json[2];
                    //**If names are returned, set the dropdown options to the returned names */
                    if (names && forms)
                        options.drugName = names.map((name, nameIndex) => {
                            return {
                                displayName: name,
                                strengthsAndForms: forms[nameIndex].map((form, formIndex) => {
                                  return {
                                    strengthAndForm: form,
                                    rxcui: cuis[nameIndex][formIndex]
                                  };
                                })
                            };
                        });

                    this.setState({ isLoading: false, options, selected: [] });
                });
```